sensor/common/scan/scan.go

```diff
@@ -297,10 +299,23 @@ func (s *LocalScan) getPullSources(srcImage *storage.ContainerImage) []*storage.
 			continue
 		}
 
+		// This ID assignment addresses an edge case where a podspec references an image
+		// by tag (e.g. latest) and the registry contents for that tag change after the pod was
+		// created. We want to ensure that we pull metadata and layers based on the
+		// ID (digest) of the running image instead of what the tag currently represents.
+		// This condition will only be true for mirrors setup via the ImageTagMirrorSet (ITMS) CR,
+		// the other supported CRs, ImageContentSourcePolicy (ICSP) and ImageDigestMirrorSet (IDMS),
+		// will only match IF the podspec references the image by digest, in which case ID
+		// would be populated and this condition never true.
+		if img.GetId() == "" && srcImage.GetId() != "" {
```

Comment:
if img.GetId() is not `""`, that means img.GetId() equals to srcImage.GetId() already?

Reply:
Yes per current mirroring CRs.

Reply:
would it be worth just unconditionally setting `id.Id = srcImage.GetId()`?

Reply:
I agree with @RTann suggestion.

Reply:
This was meant to be defensive - with the code as is we 'prefer' the output from the openshift library we use to build the mirror list (vs. overwriting it as suggested) 

After reviewing the library code, the blind assignment is likely OK - however I'm leaning toward keeping the condition since it is not complex and is no apparent harm to it.  Let me know if you feel strongly the opposite.

Reply:
sounds good to me, though curious: what happens when they are both populated but are not equal? What does this imply?

Reply:
💥 /EOL

That would imply that `c/Image`changed the digest for some unknown reason (currently wouldn't happen).